lib/streamlit/elements/button.py

```diff
@@ -92,6 +98,89 @@ def button(
             kwargs=kwargs,
         )
 
+    def download_button(
+        self,
+        label: str,
+        data: DownloadButtonDataType,
+        file_name: Optional[str] = None,
+        mime: Optional[str] = None,
+        key: Optional[Key] = None,
+        help: Optional[str] = None,
+        on_click: Optional[WidgetCallback] = None,
+        args: Optional[WidgetArgs] = None,
+        kwargs: Optional[WidgetKwargs] = None,
+    ) -> bool:
+        """Display a download button widget.
+
+        Parameters
+        ----------
+        label : str
+            A short label explaining to what this button is for.
+        data : str or bytes or file
+            The contents of the file to be downloaded.
+        file_name: str
+            An optional string to use as the name of the file to be downloaded,
+            eg. 'my_file.csv'. If file name is not specified, then we provide
+            a generic name for the download.
+        mime : str or None
+            The MIME type of the data. If None, defaults to "text/plain" or
+            "application/octet-stream" depending on the data type.
+        key : str or int
+            An optional string or integer to use as the unique key for the widget.
+            If this is omitted, a key will be generated for the widget
+            based on its content. Multiple widgets of the same type may
+            not share the same key.
```

Comment:
nit: I think this should now be "Multiple widgets may not share the same key". (And if it's not like this elsewhere, those docstrings should be updated too, but that's out of the scope of this PR)